fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb

```diff
@@ -43,8 +42,10 @@ def run!
         collector.did_raise_error(@program[:name])
         display_user_error!(e, e.message)
       rescue => e # high chance this is actually FastlaneCore::Interface::FastlaneCrash, but can be anything else
-        collector.did_raise_error(@program[:name])
+        collector.did_crash(@program[:name])
         handle_unknown_error!(e)
+      ensure
+        collector.did_finish
```

Comment:
This is code that we added last week as part of collecting tool launches, and this part was not working the way we needed it to.

The current situation is:

1. You run `fastlane beta`
1. We note that `fastlane` has launched
1. The lane calls `pilot`
1. We note that `pilot` has launched
1. `pilot` fails
1. We note that `pilot` fails and re-raise the exception
1. Now we jump from `run_active_command` straight to the error handling blocks in `fastlane_runner.rb`, so we never note that `fastlane` has finished

The end result is that the code never sends info about `fastlane` tool runs that end in an error. The change fixes it so that we will!